adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,1053 @@
+
+* [code]#accessor#: The command must not begin executing until the
+  [code]#buffer# associated with the [code]#accessor# can be accessed in a
+  manner compatible with the specified [code]#access_mode#.
+  The [code]#accessor# must have an [code]#AccessTarget# of
+  [code]#target::device# or [code]#target::host_task#.
+
+* [code]#tracking#: The command must be submitted such that its status can be
+  tracked via an [code]#event# when the [code]#tracking# object is constructed
+  with an [code]#enabled# value of [code]#true#.
+
+* [code]#kernel_bundle<bundle_state::executable>#: The command must be submitted
+  using a <<device image>> from the kernel bundle.
+
+'''
+
+.[apititle]#Default constructor#
+[source,role=synopsis,id=api:requirements-ctor]
+----
+template <typename... Requirements>
+requirements(Requirements... values);
+----
+
+_Constraints_:
+
+* [code]#is_requirement_v# returns [code]#true# for each type in
+  [code]#Requirements#;
+* [code]#Requirements# contains at most one
+  [code]#kernel_bundle<bundle_state::executable>#; and
+* [code]#Requirements# contains at most one [code]#tracking#.
+
+_Effects_: Constructs a [code]#requirements# object representing the set of
+requirements specified via the [code]#values# parameter pack.
+
+_Remarks_: Unless otherwise specified, if an instance of a requirement appears
+more than once in the [code]#values# parameter pack, the [code]#requirements#
+object behaves as if it had only been specified once.
+
+'''
+
+.[apititle]#requirements::register-accessors#
+[source,role=synopsis,id=api:register-accessors]
+----
+void register-accessors(handler& h) const;
+----
+
+This function is exposition only.
+It is shown only to help specify the effect of the functions below under "New
+free functions".
+
+_Effects_: Calls [code]#h.require# for each [code]#accessor# passed to the
+constructor of this [code]#requirements# object.
+
+'''
+
+.[apititle]#requirements::register-events#
+[source,role=synopsis,id=api:register-events]
+----
+void register-events(handler& h) const;
+----
+
+This function is exposition only.
+It is shown only to help specify the effect of the functions below under "New
+free functions".
+
+_Effects_: Calls [code]#h.depends_on# for each [code]#event# or
+[code]#std::vector<event># passed to the constructor of this
+[code]#requirements# object.
+
+'''
+
+.[apititle]#requirements::register-kernel-bundle#
+[source,role=synopsis,id=api:register-kernel-bundle]
+----
+void register-kernel-bundle(handler& h) const;
+----
+
+This function is exposition only.
+It is shown only to help specify the effect of the functions below under "New
+free functions".
+
+_Effects_: Calls [code]#h.use_kernel_bundle# if a [code]#kernel_bundle# in
+executable state was passed to the constructor of this [code]#requirements#
+object and has no effect otherwise.
+
+'''
+
+.[apititle]#requirements::has-tracking#
+[source,role=synopsis,id=api:has-tracking]
+----
+bool has-tracking() const;
```

Comment:
```suggestion
bool has_tracking() const;
```

Reply:
I think the dash is intended.  John named all the exposition-only functions with dashes to help emphasize that they are not part of the specified API.